model/task/db.go

```diff
@@ -2602,7 +2602,16 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 		{"$project": projectOut},
 		{"$match": match},
 	}
-
+	if !opts.IncludeExecutionTasks {
```

Comment:
Instead of multiple consecutive match statements in the pipeline I think we can just absorb this into the above `match` variable